package/src/components/ImageGallery/components/AnimatedGalleryVideo.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from 'react';
+import { StyleSheet, View, ViewStyle } from 'react-native';
+import type { StyleProp } from 'react-native';
+import Animated, { useAnimatedStyle } from 'react-native-reanimated';
+
+import {
+  PlaybackStatus,
+  Video,
+  VideoPayloadData,
+  VideoProgressData,
+  VideoType,
+} from '../../../native';
+
+import { vw } from '../../../utils/utils';
+import { Spinner } from '../../Spinner/Spinner';
+
+const screenWidth = vw(100);
+const halfScreenWidth = vw(50);
+const oneEight = 1 / 8;
+
+type Props = {
+  handleEnd: () => void;
+  handleLoad: (payload: VideoPayloadData) => void;
+  handleProgress: (data: VideoProgressData) => void;
+  index: number;
+  offsetScale: Animated.SharedValue<number>;
+  paused: boolean;
+  previous: boolean;
+  scale: Animated.SharedValue<number>;
+  screenHeight: number;
+  selected: boolean;
+  shouldRender: boolean;
+  source: { uri: string };
+  translateX: Animated.SharedValue<number>;
+  translateY: Animated.SharedValue<number>;
+  videoRef: React.RefObject<VideoType>;
+  style?: StyleProp<ViewStyle>;
+};
+
+const styles = StyleSheet.create({
+  activityIndicator: {
+    alignSelf: 'center',
+  },
+  videoPlayer: {
+    height: '100%',
+    width: '100%',
+  },
+});
+
+export const AnimatedGalleryVideo: React.FC<Props> = React.memo(
+  (props) => {
+    const [opacity, setOpacity] = useState<number>(1);
+    const {
+      handleEnd,
+      handleLoad,
+      handleProgress,
+      index,
+      offsetScale,
+      paused,
+      previous,
+      scale,
+      screenHeight,
+      selected,
+      shouldRender,
+      source,
+      style,
+      translateX,
+      translateY,
+      videoRef,
+    } = props;
+
+    const onLoadStart = () => {
+      setOpacity(1);
+    };
+
+    const onLoad = (payload: VideoPayloadData) => {
+      setOpacity(0);
+      handleLoad(payload);
+    };
+
+    const onEnd = () => {
+      handleEnd();
+    };
+
+    const onProgress = (data: VideoProgressData) => {
+      handleProgress(data);
+    };
+
+    const onBuffer = ({ isBuffering }: { isBuffering: boolean }) => {
+      if (isBuffering) setOpacity(1);
+      else setOpacity(0);
+    };
+
+    const onPlayBackStatusUpdate = (playbackStatus: PlaybackStatus) => {
+      if (!playbackStatus.isLoaded) {
+        // Update your UI for the unloaded state
+        setOpacity(1);
+        if (playbackStatus.error) {
+          console.log(`Encountered a fatal error during playback: ${playbackStatus.error}`);
```

Comment:
If it's a fatal error, that sounds like it should kill the app - is that right? Or should we just log it as `console.error`?